hack/test/unit-run.go

```diff
@@ -0,0 +1,220 @@
+///usr/bin/env true; exec /usr/bin/env go run "$0" "$@"
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+func init() {
+	log.SetFlags(0)
+}
+
+// envOrDefault returns the value of environment variable env if it is set and def otherwise.
+func envOrDefault(env, def string) string {
+	if s, ok := os.LookupEnv(env); ok {
+		return s
+	}
+	return def
+}
+
+// partitionStrings partitions strings based on the general predicate p.
+func partitionStrings(p func(s string) bool, ss ...string) (t, f []string) {
+	t, f = ss[:0:0], ss[:0:0]
+	for _, s := range ss {
+		if p(s) {
+			t = append(t, s)
+		} else {
+			f = append(f, s)
+		}
+	}
+	return t, f
+}
+
+// filterStrings filters strings based on the general predicate p.
+func filterStrings(p func(s string) bool, ss ...string) []string {
+	t := ss[:0:0]
+	for _, s := range ss {
+		if p(s) {
+			t = append(t, s)
+		}
+	}
+	return t
+}
+
+// pkgPath returns fully-qualified Go package name given a directory relative to repository root.
+func pkgPath(dir string) string {
+	const repo = "github.com/docker/docker"
+	if dir == "" {
+		return repo
+	}
+	return fmt.Sprintf("%s/%s", repo, dir)
+}
+
+// goTest executes `go test` over packages in pkgs.
+// goTest returns the exit code of the executed `go test` command or go execution error.
+func goTest(testFlags []string, sequential bool, coverprofile string, pkgs ...string) (int, error) {
+	args := append([]string{
+		"test",
+		"-json",
+		"-tags='netgo seccomp libdm_no_deferred_remove'",
+		"-cover",
+		"-covermode=atomic",
+		fmt.Sprintf("-coverprofile=%s", coverprofile),
+	}, testFlags...)
+	if sequential {
+		args = append(args, "-p=1")
+	}
+	cmd := exec.Command("go", append(args, pkgs...)...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	switch err := cmd.Run().(type) {
+	case nil, *exec.ExitError:
+		return cmd.ProcessState.ExitCode(), nil
+	default:
+		return 0, err
+	}
+}
+
+// mergeProfiles writes profile lines from each profile in profiles to final test profile at path.
+// mergeProfiles assumes that a valid profile file already exists at path.
+func mergeProfiles(path string, profiles ...*os.File) error {
```

Comment:
AFAIK, profiles `go` testing framework outputs are sorted. This implementation does not provide such guarantees. Whether or not this is a problem for our tooling - I am not sure. Can someone confirm that simply appending is OK?